SimPEG/FLOW/Richards/Empirical.py

```diff
@@ -1,450 +1,656 @@
+        ddm[u >= 0] = 0
+        dn = Utils.sdiag(ddm) * self.nDeriv
+        return dn
+
+    def _derivI(self, u):
+        if self.IMap is None:
+            return Utils.Zero()
+        Ks, alpha, I, n, m = self._get_params()
+        ddm = Ks*(1.0*(abs(alpha*u)**n + 1.0)**(-1.0 + 1.0/n))**I*(-(-(1.0*(abs(alpha*u)**n + 1.0)**(-1.0 + 1.0/n))**(1.0/(1.0 - 1.0/n)) + 1.0)**(1.0 - 1.0/n) + 1.0)**2*np.log(1.0*(abs(alpha*u)**n + 1.0)**(-1.0 + 1.0/n))
+        ddm[u >= 0] = 0
+        dI = Utils.sdiag(ddm) * self.IDeriv
+        return dI
+
+    def derivU(self, u):
+        Ks, alpha, I, n, m = self._get_params()
+        ddm = I*alpha*n*Ks*abs(alpha*u)**(n - 1.0)*np.sign(alpha*u)*(1.0/n - 1.0)*((abs(alpha*u)**n + 1)**(1.0/n - 1))**(I - 1)*((1 - 1.0/((abs(alpha*u)**n + 1)**(1.0/n - 1))**(1.0/(1.0/n - 1)))**(1 - 1.0/n) - 1)**2*(abs(alpha*u)**n + 1)**(1.0/n - 2) - (2*alpha*n*Ks*abs(alpha*u)**(n - 1)*np.sign(alpha*u)*(1.0/n - 1)*((abs(alpha*u)**n + 1)**(1.0/n - 1))**I*((1 - 1.0/((abs(alpha*u)**n + 1)**(1.0/n - 1))**(1.0/(1.0/n - 1)))**(1 - 1.0/n) - 1)*(abs(alpha*u)**n + 1)**(1.0/n - 2))/(((abs(alpha*u)**n + 1)**(1.0/n - 1))**(1.0/(1.0/n - 1) + 1)*(1 - 1.0/((abs(alpha*u)**n + 1)**(1.0/n - 1))**(1.0/(1.0/n - 1)))**(1.0/n))
+        ddm[u >= 0] = 0
+        g = Utils.sdiag(ddm)
+        return g
 
-class VanGenuchtenParams(object):
-    """
-        The RETC code for quantifying the hydraulic functions of unsaturated soils,
-        Van Genuchten, M Th, Leij, F J, Yates, S R
 
-        Table 3: Average values for selected soil water retention and hydraulic
-        conductivity parameters for 11 major soil textural groups
-        according to Rawls et al. [1982]
+def van_genuchten(mesh, **kwargs):
+    return _partition_args(
+        mesh,
+        Vangenuchten_k,
+        Vangenuchten_theta,
+        ['alpha', 'n', 'Ks', 'I'],
+        ['alpha', 'n', 'theta_r', 'theta_s'],
+        **kwargs
+    )
 
+
+class VanGenuchtenParams(object):
+    """The RETC code for quantifying the hydraulic functions of unsaturated
+    soils, Van Genuchten, M Th, Leij, F J, Yates, S R
+
+    Table 3: Average values for selected soil water retention and hydraulic
+    conductivity parameters for 11 major soil textural groups
+    according to Rawls et al. [1982]
     """
-    def __init__(self): pass
+
     @property
     def sand(self):
-        return {"theta_r": 0.020, "theta_s": 0.417, "alpha": 0.138*100., "n": 1.592, "Ks": 504.0/100./24./60./60.}
+        return {
+            "theta_r": 0.020, "theta_s": 0.417, "alpha": 0.138*100.,
+            "n": 1.592, "Ks": 504.0/100./24./60./60.
+        }
```

Comment:
I would like to plug scipy's little used "constants" package for this, makes things nice and explicit when you have units, and converts everything to MKS types.